packages/components/src/test/table/UsePresentationTable.test.tsx

```diff
@@ -223,24 +218,113 @@ describe("usePresentationTableWithUnifiedSelection", () => {
     expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
   });
 
-  it("returns an array of selectedRows when keys are passed correctly on selectionChange event", async () => {
+  it("returns an array of selectedRows when keys are passed before the table is loaded", async () => {
     const keys = new KeySet([createTestECInstanceKey()]);
     sinon.stub(Presentation.selection, "getSelection").returns(new KeySet([createTestECInstanceKey()]));
 
-    const { descriptor, item } = getMockData();
-    presentationManagerMock.setup(async (x) => x.getContentDescriptor(moq.It.is((options) => options.keys.size === keys.size))).returns(async () => descriptor);
-    presentationManagerMock
-      .setup(async (x) => x.getContentAndSize(moq.It.is((options) => options.keys.size === keys.size)))
-      .returns(async () => ({ content: new Content(descriptor, [item]), size: 1 }));
+    setupPresentationManager();
+    Presentation.selection.addToSelection("UnifiedSelectionContext", initialProps.imodel, keys, 1);
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    const resultAfterLoading = result.current.selectedRows;
+    expect(resultAfterLoading.length).to.be.equal(1);
+  });
+
+  it("returns an array of selectedRows when keys are added on selectionChange event", async () => {
+    const instanceKey1 = createTestECInstanceKey();
+    const instanceKey2 = createTestECInstanceKey({ className: "TestSchema:TestClass", id: "0x2" });
+    const keySet1 = new KeySet([instanceKey1]);
+    const keySet2 = new KeySet([instanceKey2]);
+    const keySet3 = new KeySet([instanceKey1, instanceKey2]);
+
+    setupPresentationManager(2);
+    Presentation.selection.addToSelection("UnifiedSelectionContext", imodel, keySet3, 0);
+    Presentation.selection.addToSelection("UnifiedSelectionContext", imodel, keySet1, 1);
```

Comment:
- [ ] `id` of `instanceKey1` is not clear here - I recommend specifying it explicitly, even if it matches the default value.
- [ ] For `instanceKey2` you probably don't need to specify a class.
- [ ] I don't think you need to create these `KeySet` objects here. They're all used only once, their name tells nothing and they only make it harder to read the code - whenever this object I used I have to look up the declaration to see what's stored in there. Instead, I recommend just creating those keysets where you use them.
- [ ] It's not clear what instances `setupPresentationManager` creates rows for. I recommend changing it to take a list of instance keys rather than a number.
- [ ] Seeing two calls to `Presentation.selection.addToSelection` is a bit confusing - can you add comments why you call each of them? E.g.
   ```ts
   // select both instances at level 0 to get them displayed in the component
   Presentation.selection.addToSelection("UnifiedSelectionContext", imodel, new KeySet([ instanceKey1, instanceKey2 ]), 0);
   // select instanceKey1 at level 1 to get its row selected 
   Presentation.selection.addToSelection("UnifiedSelectionContext", imodel, new KeySet([ instanceKey1 ]), 1);
   ```